lint-rules.md

```diff
@@ -0,0 +1,1266 @@
+Closing brackets (either `)` or `]`) should **not** be **preceeded** by whitespace **unless** that whitespace is **indentation**.
+
+Sometimes a long expression will exceed the maximum line width. In these cases, one or more linebreaks should be introduced. Line continuations should be indented one more level than the beginning of the expression. If bracketed content (things between `()` or `[]`) must be split onto multiple lines, a newline should follow the opening bracket, the contents should be indented an additional level, then the closing bracket should be de-indented to match the indentation of the opening bracket.
+
+If you are line splitting an expression on an infix operator, the operator should be on the continued line.
+
+If you are using the `if...then...else` construct as part of your expression and it needs to be line split, the entire construct should be wrapped in parantheses (`()`). The opening paranthesis should be immediately followed by a newline. `if`, `then`, and `else` should all start a line one more level of indentation than the wrapping paratheses. The closing paranthesis should be on the same level of indentation as the opening paranthesis.
+If you are using the `if...then...else` construct on one line, it does not need to be wrapped in parantheses. However, if any of the 3 clauses are more complex than a single identifier, they should be wrapped in parantheses.
+
+Sometimes a developer will choose to line split an expression despite it being able to all fit on one line that is <=90 characters wide. That is perfectly acceptable, though you may notice in the below examples the single line form can be more readable.
+
+**Group**: `spacing`
+
+#### Example
+
+Complex example that fits on one line:
+
+```wdl
+Int complex_value = w - x + (y * (z / (foo % bar)))
+```
+
+Same example with line splitting:
+
+```wdl
+Int complex_value = w
+    - x
+    + (
+        y
+        * (
+            z
+            / (
+                foo % bar
+            )
+        )
+    )
+```
+
+Another complex example:
+
+```wdl
+Boolean complicated_logic = (
+    if !(
+        a
+        && b
+        || c
+        && (
+            !d || !e
+        )
+        == (
+            if foobar
+            then gak
+            else caz
+        )
+    )
+    then "wow"
+    else "WOWOWOW"
+)
+```
+
+### `input_spacing`
+
+When making calls from a workflow, it is more readable and easier to edit if the supplied inputs are each on their own line. This does inflate the line count of a WDL dcoument, but it is worth it for the consistent readability. An exception _can_ be made (but does not have to be made), for calls with only a single parameter. In those cases, it is permissable to keep the input on the same line as the call.
+
+When newline separating inputs to a call, always follow each input with a trailing comma. See examples below.
```

Comment:
Can you elaborate? I have reasons for preferring it, but I'm not above having my mind changed on the matter.

My reasons for being pro-trailing-comma
* Cleaner diffs
    * New values can be appended to a list "cleanly"
    * No trailing comma, and the former-last value will appear "changed" in the diff as a comma must be appended to an otherwise unchanging line
* In my experience, trailing commas are easier to stay compliant with when changing an existing list. I've definitely been burned by compilers that complained about a missing comma when I add to a list. Or conversely in languages where trailing commas are disbarred (JSON) been burned by deleting the last element and forgetting to clean up the now-trailing-comma that must be deleted. IMO it's just "easier" to always have a comma at the end. Very subjective point here. Not my strongest argument, but I stand by it lol.

Reply:
I don't feel strongly enough about this, particularly if we support multiple options (as discussed via slack). I agree with you from the perspective of `diff`. However, for reading, I think omitting the trailing comma (as you would in standard written English) is cleaner. When I see a list with a trailing comma, I want to verify that something wasn't omitted from the list accidentally. I also think that a trailing comma encourages appending, which is not always the right choice when adding to a list.